beacon-chain/p2p/pubsub.go

```diff
@@ -47,7 +47,19 @@ func (s *Service) PublishToTopic(ctx context.Context, topic string, data []byte,
 	if err != nil {
 		return err
 	}
-	return topicHandle.Publish(ctx, data, opts...)
+
+	// Wait for at least 1 peer to be available to receive the published message.
+	for {
```

Comment:
Can we make sure all callers of this are in a separate go-routine ? So that in the event it is blocked here, it doesn't block any other important functions that need to be called.